### ISSUE TITLE
Debugger hits SIGSEGV in ShimStackWalk::StackWalkInfo::GetCurrentInternalFrame when .NET app throws exception

### ISSUE THREAD

**Issue by mikelle-rogers** [dotnet/runtime#105374]
### Description

With debug builds of the runtime, an assert is hit in `ShimStackWalk::StackWalkInfo::GetCurrentInternalFrame`. 

With release a release build of the runtime, we hit a signal SIGSEGV: invalid address (fault address: 0x0) in `ShimStackWalk::StackWalkInfo::GetCurrentInternalFrame`

### Reproduction Steps

1. Create a console application with this code.

```
using Azure.Identity;
using Azure.Core;

var cred = new DefaultAzureCredential();

string[] scopes = new[] { "https://management.azure.com/.default" };
var token = new TokenRequestContext(scopes);

cred.GetToken(token, CancellationToken.None);

Console.WriteLine("Foo");
```

2. Put a breakpoint on the last line, it won't be hit
3. If you attach to vsdbg-ui while on the `cred.GetToken` line, and the Runtime is built in debug, you will hit the assert. 

### Expected behavior

The debugger does not crash and the breakpoint on the last line is hit.

### Actual behavior

Breakpoint on last line is not hit because the debugger crashes.

### Configuration

WSL, .NET 9 preview 6
Repros with DOTNET_LegacyExceptionHandling = 1 or 0

### Other information

vsdbg-ui call stack with debug runtime
```
* thread #14, name = 'XAPI-Worker', stop reason = signal SIGTRAP
    frame #0: 0x00007f4299b7cfad libmscordaccore.so`DBG_DebugBreak at debugbreak.S:9
   6
   7    LEAF_ENTRY DBG_DebugBreak, _TEXT
   8            int3
-> 9            ret
   10   LEAF_END_MARKED DBG_DebugBreak, _TEXT
   11
(lldb) bt
* thread #14, name = 'XAPI-Worker', stop reason = signal SIGTRAP
  * frame #0: 0x00007f4299b7cfad libmscordaccore.so`DBG_DebugBreak at debugbreak.S:9
    frame #1: 0x00007f4299ae796b libmscordaccore.so`::DebugBreak() at debug.cpp:407:9
    frame #2: 0x00007f4299ec9653 libmscordbi.so`::DbgAssertDialog(szFile="/home/mirogers/Microsoft/runtime/src/coreclr/debug/di/shimstackwalk.cpp", iLine=1547, szExpr="!ExhaustedAllInternalFrames() || HasConvertedFrame()") at debug.cpp:458:13
    frame #3: 0x00007f4299d01f54 libmscordbi.so`ShimStackWalk::StackWalkInfo::GetCurrentInternalFrame(this=0x00007f42b0ba2d20) at shimstackwalk.cpp:1547:5
    frame #4: 0x00007f4299d01c8a libmscordbi.so`ShimStackWalk::ShouldTrackUMChain(this=0x00007f42a07695d0, pswInfo=0x00007f42b0ba2d20) at shimstackwalk.cpp:158:39
    frame #5: 0x00007f4299d011c2 libmscordbi.so`ShimStackWalk::Populate(this=0x00007f42a07695d0) at shimstackwalk.cpp:417:21
    frame #6: 0x00007f4299d0022a libmscordbi.so`ShimStackWalk::ShimStackWalk(this=0x00007f42a07695d0, pProcess=0x00007f4290000b80, pThread=0x00007f427c02f4e0) at shimstackwalk.cpp:33:5
    frame #7: 0x00007f4299cf1ab9 libmscordbi.so`ShimProcess::LookupOrCreateShimStackWalk(this=0x00007f4290000b80, pThread=0x00007f427c02f4e0) at shimprocess.cpp:1490:45
    frame #8: 0x00007f4299e1be91 libmscordbi.so`CordbThread::EnumerateChains(this=0x00007f427c02f4e0, ppChains=0x00007f42b0ba3a80) at rsthread.cpp:970:64
   ...
```

vsdbg-ui call stack with release runtime
```
* thread #14, name = 'XAPI-Worker', stop reason = signal SIGSEGV: invalid address (fault address: 0x0)
  * frame #0: 0x00007f421b7472f2 libmscordbi.so`ShimStackWalk::Populate() [inlined] ShimStackWalk::StackWalkInfo::GetCurrentInternalFrame(this=0x00007f423e098ed0) at shimstackwalk.cpp:0 [opt]
    frame #1: 0x00007f421b7472d6 libmscordbi.so`ShimStackWalk::Populate() [inlined] ShimStackWalk::ShouldTrackUMChain(this=0x00007f4224423660, pswInfo=0x00007f423e098ed0) at shimstackwalk.cpp:158 [opt]
    frame #2: 0x00007f421b747292 libmscordbi.so`ShimStackWalk::Populate(this=0x00007f4224423660) at shimstackwalk.cpp:417 [opt]
    frame #3: 0x00007f421b746aa2 libmscordbi.so`ShimStackWalk::ShimStackWalk(this=0x00007f4224423660, pProcess=0x00007f422000c510, pThread=0x00007f421001d770) at shimstackwalk.cpp:33:5 [opt]
    frame #4: 0x00007f421b707d58 libmscordbi.so`ShimProcess::LookupOrCreateShimStackWalk(this=0x00007f422000c510, pThread=0x00007f421001d770) at shimprocess.cpp:1490:45 [opt]
    frame #5: 0x00007f421b75017e libmscordbi.so`CordbThread::EnumerateChains(this=<unavailable>, ppChains=0x00007f423e099a80) at rsthread.cpp:970:64 [opt]
...
```

App call stack
```
* thread #1, name = 'brokenAuthNET9', stop reason = signal SIGSTOP
  * frame #0: 0x00007fee3ad27376 libpthread.so.0`__pthread_cond_wait at futex-internal.h:183:13
    frame #1: 0x00007fee3ad27359 libpthread.so.0`__pthread_cond_wait at pthread_cond_wait.c:508
    frame #2: 0x00007fee3ad27290 libpthread.so.0`__pthread_cond_wait(cond=0x000055563f1586e8, mutex=0x000055563f1586c0) at pthread_cond_wait.c:647
    frame #3: 0x00007fee3a6371a2 libcoreclr.so`CorUnix::CPalSynchronizationManager::ThreadNativeWait(ptnwdNativeWaitData=0x000055563f1586c0, dwTimeout=<unavailable>, ptwrWakeupReason=0x00007ffe9c9246ec, pdwSignaledObject=0x00007ffe9c9246e8) at synchmanager.cpp:479:28 [opt]
    frame #4: 0x00007fee3a636daf libcoreclr.so`CorUnix::CPalSynchronizationManager::BlockThread(this=0x000055563f15ca70, pthrCurrent=0x000055563f158520, dwTimeout=4294967295, fAlertable=false, fIsSleep=<unavailable>, ptwrWakeupReason=0x00007ffe9c924788, pdwSignaledObject=0x00007ffe9c92478c) at synchmanager.cpp:302:22 [opt]
    frame #5: 0x00007fee3a63b97e libcoreclr.so`CorUnix::InternalWaitForMultipleObjectsEx(pThread=0x000055563f158520, nCount=1, lpHandles=<unavailable>, bWaitAll=<unavailable>, dwMilliseconds=4294967295, bAlertable=NO, bPrioritize=NO) at wait.cpp:637:45 [opt]
    frame #6: 0x00007fee3a63bb59 libcoreclr.so`::WaitForSingleObjectEx(hHandle=0x0000000000000070, dwMilliseconds=4294967295, bAlertable=NO) at wait.cpp:138:13 [opt]
    frame #7: 0x00007fee3a398d4e libcoreclr.so`CLREventBase::WaitEx(unsigned int, WaitMode, PendingSync*) [inlined] CLREventWaitHelper2(handle=<unavailable>, dwMilliseconds=<unavailable>, alertable=<unavailable>) + 238 [opt]
    frame #8: 0x00007fee3a398d49 libcoreclr.so`CLREventBase::WaitEx(unsigned int, WaitMode, PendingSync*) [inlined] CLREventWaitHelper(pParam=0x00007ffe9c924960)::$_1::operator()(CLREventWaitHelper(void*, unsigned int, int)::Param*) const [opt]
    frame #9: 0x00007fee3a398d40 libcoreclr.so`CLREventBase::WaitEx(unsigned int, WaitMode, PendingSync*) [opt]
    frame #10: 0x00007fee3a398cdb libcoreclr.so`CLREventBase::WaitEx(this=0x000055563f1a3520, dwMilliseconds=<unavailable>, mode=WaitMode_None, syncState=0x0000000000000000) [opt]
    frame #11: 0x00007fee3a39dffb libcoreclr.so`Thread::WaitSuspendEventsHelper(this=<unavailable>) + 267 [opt]
    frame #12: 0x00007fee3a39c888 libcoreclr.so`Thread::RareDisablePreemptiveGC() + 21 [opt]
    frame #13: 0x00007fee3a39c873 libcoreclr.so`Thread::RareDisablePreemptiveGC(this=0x000055563f1a3440) [opt]
    frame #14: 0x00007fee3a51c451 libcoreclr.so`Debugger::SendExceptionHelperAndBlock(Thread*, OBJECTHANDLE__*, bool, FramePointer, unsigned long, CorDebugExceptionCallbackType, unsigned int) [inlined] GCHolderEEInterface<0, 1, 1>::LeaveInternal(this=0x00007ffe9c924abe) + 46 [opt]
    frame #15: 0x00007fee3a51c423 libcoreclr.so`Debugger::SendExceptionHelperAndBlock(Thread*, OBJECTHANDLE__*, bool, FramePointer, unsigned long, CorDebugExceptionCallbackType, unsigned int) [inlined] GCHolderEEInterface<0, 1, 1>::~GCHolderEEInterface(this=0x00007ffe9c924abe) [opt]
    frame #16: 0x00007fee3a51c423 libcoreclr.so`Debugger::SendExceptionHelperAndBlock(this=0x000055563f18da80, pThread=0x000055563f1a3440, exceptionHandle=0x00007fee38611388, fContinuable=true, framePointer=(m_sp = 0x0000000000000000), nOffset=18446744073709551615, eventType=DEBUG_EXCEPTION_UNHANDLED, dwFlags=1) [opt]
    frame #17: 0x00007fee3a51ca6f libcoreclr.so`Debugger::SendExceptionEventsWorker(this=0x000055563f18da80, pThread=0x000055563f1a3440, fFirstChance=false, fIsInterceptable=true, fContinuable=true, currentIP=<unavailable>, framePointer=(m_sp = 0x00007ffe9c9269f0), atSafePlace=<unavailable>) + 1391 [opt]
    frame #18: 0x00007fee3a51ce47 libcoreclr.so`Debugger::SendException(this=0x000055563f18da80, pThread=0x000055563f1a3440, fFirstChance=false, currentIP=140659037483225, currentSP=140731525261808, fContinuable=true, fAttaching=<unavailable>, fForceNonInterceptable=<unavailable>, pExceptionInfo=0x00007ffe9c926880) + 519 [opt]
    frame #19: 0x00007fee3a51eaba libcoreclr.so`Debugger::LastChanceManagedException(this=0x000055563f18da80, pExceptionInfo=0x00007ffe9c926880, pThread=<unavailable>, jitAttachRequested=NO) + 810 [opt]
    frame #20: 0x00007fee3a1f2ddf libcoreclr.so`NotifyDebuggerLastChance(pThread=0x000055563f1a3440, pExceptionInfo=0x00007ffe9c926880, jitAttachRequested=<unavailable>) + 111 [opt]
    frame #21: 0x00007fee3a1f3de0 libcoreclr.so`InternalUnhandledExceptionFilter_Worker(_EXCEPTION_POINTERS*) + 527 [opt]
    frame #22: 0x00007fee3a1f3bd1 libcoreclr.so`InternalUnhandledExceptionFilter_Worker(pExceptionInfo=<unavailable>) [opt]
    frame #23: 0x00007fee3a3c6b74 libcoreclr.so`UnwindManagedExceptionPass1(ex=0x00007ffe9c926880, frameContext=0x00007ffe9c925c00) + 1060 [opt]
    frame #24: 0x00007fee3a3c6bec libcoreclr.so`DispatchManagedException(ex=0x00007ffe9c926880, isHardwareException=<unavailable>) + 108 [opt]
    frame #25: 0x00007fee3a32caa4 libcoreclr.so`IL_Throw(obj=<unavailable>) + 692 [opt]
    frame #26: 0x00007fedbbf6b0d9
    frame #27: 0x00007fedbb2b9c22
    frame #28: 0x00007fedbb2b9b81
    frame #29: 0x00007fedbbf6be02
    frame #30: 0x00007fedbbf6bd56
    frame #31: 0x00007fedbbf6bc43
    frame #32: 0x00007fedbbf64eed
    frame #33: 0x00007fedbbf519b8
    frame #34: 0x00007fee3a48e48c libcoreclr.so`CallDescrWorkerInternal at calldescrworkeramd64.S:70
    frame #35: 0x00007fee3a2ca92c libcoreclr.so`MethodDescCallSite::CallTargetWorker(unsigned long const*, unsigned long*, int) + 89 [opt]
    frame #36: 0x00007fee3a2ca8d3 libcoreclr.so`MethodDescCallSite::CallTargetWorker(this=0x00007fada696ece8, pArguments=0x00007ffe9c926f48, pReturnValue=0x0000000000000000, cbReturnValue=0) [opt]
    frame #37: 0x00007fee3a1b17d4 libcoreclr.so`RunMain(MethodDesc*, short, int*, PtrArray**) [inlined] MethodDescCallSite::Call(this=0x00007ffe9c926fa8, pArguments=0x00007ffe9c926f48) + 9 [opt]
    frame #38: 0x00007fee3a1b17cb libcoreclr.so`RunMain(MethodDesc*, short, int*, PtrArray**) [opt]
    frame #39: 0x00007fee3a1b1679 libcoreclr.so`RunMain(MethodDesc*, short, int*, PtrArray**) [inlined] RunMain(pParam=0x00007ffe9c926f10)::$_0::operator()(Param*) const::'lambda'(Param*)::operator()(Param*) const [opt]
    frame #40: 0x00007fee3a1b1679 libcoreclr.so`RunMain(MethodDesc*, short, int*, PtrArray**) [opt]
    frame #41: 0x00007fee3a1b166c libcoreclr.so`RunMain(pFD=0x00007fedbbfe3a08, numSkipArgs=1, piRetVal=<unavailable>, stringArgs=<unavailable>) [opt]
    frame #42: 0x00007fee3a1b1c4c libcoreclr.so`Assembly::ExecuteMainMethod(this=0x000055563f120530, stringArgs=0x00007ffe9c9273d0, waitForOtherThreads=YES) + 460 [opt]
    frame #43: 0x00007fee3a1db2df libcoreclr.so`CorHost2::ExecuteAssembly(this=<unavailable>, dwAppDomainId=<unavailable>, pwzAssemblyPath=<unavailable>, argc=0, argv=0x0000000000000000, pReturnValue=0x00007ffe9c927540) + 735 [opt]
    frame #44: 0x00007fee3a19f1a5 libcoreclr.so`::coreclr_execute_assembly(hostHandle=0x000055563f15e480, domainId=1, argc=0, argv=<unavailable>, managedAssemblyPath=<unavailable>, exitCode=0x00007ffe9c927540) at exports.cpp:494:24 [opt]
    frame #45: 0x00007fee3a76b2d1 libhostpolicy.so`___lldb_unnamed_symbol213$$libhostpolicy.so + 1089
    frame #46: 0x00007fee3a76c3c9 libhostpolicy.so`corehost_main + 345
    frame #47: 0x00007fee3a7ab535 libhostfxr.so`___lldb_unnamed_symbol78$$libhostfxr.so + 1477
    frame #48: 0x00007fee3a7aa52d libhostfxr.so`___lldb_unnamed_symbol76$$libhostfxr.so + 765
    frame #49: 0x00007fee3a7a44a2 libhostfxr.so`hostfxr_main_startupinfo + 242
    frame #50: 0x000055563dc34b0b brokenAuthNET9`___lldb_unnamed_symbol13$$brokenAuthNET9 + 1131
    frame #51: 0x000055563dc34e2f brokenAuthNET9`___lldb_unnamed_symbol15$$brokenAuthNET9 + 175
    frame #52: 0x00007fee3a7fe083 libc.so.6`__libc_start_main(main=(brokenAuthNET9`___lldb_unnamed_symbol15$$brokenAuthNET9), argc=1, argv=0x00007ffe9c927c48, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007ffe9c927c38) at libc-start.c:308:16
    frame #53: 0x000055563dc34079 brokenAuthNET9`___lldb_unnamed_symbol1$$brokenAuthNET9 + 41
```

**Comment by thaystg** [comment]
Ah, I am almost sure this one was fixed by this PR: https://github.com/dotnet/runtime/pull/106060.

Can you test and close @mikelle-rogers

**Comment by mikelle-rogers** [comment]
> Ah, I am almost sure this one was fixed by this PR: #106060.
> 
> Can you please test and close @mikelle-rogers ?

I have verified that this issue is fixed.
